src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -105,6 +105,20 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
                 server.Start(_portOptions.MaxConnections.GetValueOrDefault(ReversedDiagnosticsServer.MaxAllowedConnections));
 
+                using FileSystemWatcher watcher = new(Path.GetDirectoryName(_portOptions.EndpointName));
+                void onDiagnosticPortAltered()
+                {
+                    _logger.DiagnosticPortAlteredWhileInUse(_portOptions.EndpointName);
+                    watcher.EnableRaisingEvents = false;
+                }
+
+                watcher.Filter = Path.GetFileName(_portOptions.EndpointName);
+                watcher.NotifyFilter = NotifyFilters.FileName;
+                watcher.Deleted += (object _, FileSystemEventArgs _) => onDiagnosticPortAltered();
```

Comment:
Are the types necessary or can you discard everything?

```suggestion
                watcher.Deleted += (_,  _) => onDiagnosticPortAltered();
```